apps/protocol-frontend/src/hooks/useContributionList.ts

```diff
@@ -0,0 +1,60 @@
+import { useState } from 'react';
+import { useUser } from '../contexts/UserContext';
+import { useQuery, useInfiniteQuery } from '@tanstack/react-query';
+import { UIContribution } from '@govrn/ui-types';
+import { ListContributionsQueryVariables } from '@govrn/protocol-client';
+
+export const useContributionList = (args?: ListContributionsQueryVariables) => {
+  const { govrnProtocol: govrn } = useUser();
+  const { isLoading, isFetching, isError, error, data } = useQuery(
+    ['contributionList', args],
+    async (): Promise<UIContribution[]> => {
+      const data = await govrn.contribution.list(args || {});
+      return data.result;
+    },
+  );
+  return { isLoading, isError, isFetching, error, data };
+};
+
+export const useContributionInfiniteList = (
+  args?: ListContributionsQueryVariables,
+  pageSize = 20,
+) => {
+  const { govrnProtocol: govrn } = useUser();
+  const {
+    status,
+    data,
+    isFetching,
+    isFetchingNextPage,
+    fetchNextPage,
+    hasNextPage,
+  } = useInfiniteQuery(
+    ['contributionList', args],
```

Comment:
I'm thinking we may need the key here to be `contributionInfiniteList` although I'm not completely sure. I came across this in a blog post at some point: useQuery(['todos'], fetchTodos)

https://tkdodo.eu/blog/effective-react-query-keys#caching-data

Not sure if we need to worry about it, but figured I'd bring it up. This person recommends not mixing the keys when having an infinite scroll and a regular query.